<a href="https://colab.research.google.com/github/Hyeon-Seok/Python/blob/master/Pandas_Exam_job.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

### Step 1. 필요한 라이브러리 임포트

In [0]:
import pandas as pd

### Step 2. [데이터](https://raw.githubusercontent.com/justmarkham/DAT8/master/data/u.user) 확인

- user_id 유저 아이디
- age 나이
- gender 성별
- occupation 직업
- zip_code 우편번호

### Step 3. 

**read_csv를 통해 데이터를 올바른 형태로 불러온 뒤 users에 저장해주세요.**

user_id를 인덱스로 설정해주세요.

In [0]:
url = 'https://raw.githubusercontent.com/justmarkham/DAT8/master/data/u.user'

users = pd.read_csv(url, sep='|')
users = users.set_index('user_id')
users.head()

# users2 = pd.read_csv(url, sep='|', index_col='user_id')

age gender  occupation zip_code
user_id                                 
1         24      M  technician    85711
2         53      F       other    94043
3         23      M      writer    32067
4         24      M  technician    43537
5         33      F       other    15213

### Step 4. 초반 25행 조회

In [0]:
users.head(25)

age gender     occupation zip_code
user_id                                    
1         24      M     technician    85711
2         53      F          other    94043
3         23      M         writer    32067
4         24      M     technician    43537
5         33      F          other    15213
6         42      M      executive    98101
7         57      M  administrator    91344
8         36      M  administrator    05201
9         29      M        student    01002
10        53      M         lawyer    90703
11        39      F          other    30329
12        28      F          other    06405
13        47      M       educator    29206
14        45      M      scientist    55106
15        49      F       educator    97301
16        21      M  entertainment    10309
17        30      M     programmer    06355
18        35      F          other    37212
19        40      M      librarian    02138
20        42      F      homemaker    95660
21        26      M         writer    30068
22        25      M         writer    40206
23        30      F         artist    48197
24        21      F         artist    94533
25        39      M       engineer    55107

### Step 5. 마지막 10행 조회

In [0]:
users.tail(10)

age gender     occupation zip_code
user_id                                    
934       61      M       engineer    22902
935       42      M         doctor    66221
936       24      M          other    32789
937       48      M       educator    98072
938       38      F     technician    55038
939       26      F        student    33319
940       32      M  administrator    02215
941       20      M        student    97229
942       48      F      librarian    78209
943       22      M        student    77841

### Step 6. 데이터셋은 몇 개의 행으로 이루어져 있나요?

943

In [0]:
len(users)

943

### Step 7.  데이터셋은 몇 개의 열로 이루어져 있나요?

4

In [0]:
users.shape[1]

4

### Step 8. 열(column)의 이름들을 모두 조회해주세요

Index(['age', 'gender', 'occupation', 'zip_code'], dtype='object')

In [0]:
users.columns

Index(['age', 'gender', 'occupation', 'zip_code'], dtype='object')

### Step 9. 데이터셋의 인덱스를 조회해주세요

Int64Index([  1,   2,   3,   4,   5,   6,   7,   8,   9,  10,
            ...
            934, 935, 936, 937, 938, 939, 940, 941, 942, 943],
           dtype='int64', name='user_id', length=943)

In [0]:
users.index

Int64Index([  1,   2,   3,   4,   5,   6,   7,   8,   9,  10,
            ...
            934, 935, 936, 937, 938, 939, 940, 941, 942, 943],
           dtype='int64', name='user_id', length=943)

### Step 10. 각 열의 데이터타입은 무엇인가요?

In [0]:
users.dtypes

age            int64
gender        object
occupation    object
zip_code      object
dtype: object

### Step 11. occupation column만 조회해주세요

In [0]:
users['occupation']

user_id
1         technician
2              other
3             writer
4         technician
5              other
6          executive
7      administrator
8      administrator
9            student
10            lawyer
11             other
12             other
13          educator
14         scientist
15          educator
16     entertainment
17        programmer
18             other
19         librarian
20         homemaker
21            writer
22            writer
23            artist
24            artist
25          engineer
26          engineer
27         librarian
28            writer
29        programmer
30           student
           ...      
914            other
915    entertainment
916         engineer
917          student
918        scientist
919            other
920           artist
921          student
922    administrator
923          student
924            other
925         salesman
926    entertainment
927       programmer
928          student
929        scientist
930  

### Step 12. 데이터셋에는 몇 종류의 직업이 존재하나요?

21

In [0]:
users['occupation'].nunique()

21

### Step 13. 데이터셋에서 가장 많이 존재하는 직업은 무엇인가요?

student

In [0]:
users.groupby('occupation').count().sort_values('age', ascending=False).head(1)
# (users.groupby('occupation')['occupation'].apply(lambda x: x.value_counts().head(1))).sort_values(, ascending=False)

age  gender  zip_code
occupation                       
student     196     196       196

### Step 14. users.describe() 코드를 실행해서 숫자 열의 요약정보를 조회해주세요

In [0]:
users.describe()

### Step 15. users.describe(include = 'all') 코드를 실행해서 모든 열의 정보를 조회해주세요

In [0]:
users.describe(include = 'all')

age gender occupation zip_code
count   943.000000    943        943      943
unique         NaN      2         21      795
top            NaN      M    student    55414
freq           NaN    670        196        9
mean     34.051962    NaN        NaN      NaN
std      12.192740    NaN        NaN      NaN
min       7.000000    NaN        NaN      NaN
25%      25.000000    NaN        NaN      NaN
50%      31.000000    NaN        NaN      NaN
75%      43.000000    NaN        NaN      NaN
max      73.000000    NaN        NaN      NaN

### Step 16. occupation 칼럼만 조회해서 describe() 함수를 실행해주세요

count         943<br>
unique         21<br>
top       student<br>
freq          196<br>
Name: occupation, dtype: object<br>

In [0]:
users['occupation'].describe()

count         943
unique         21
top       student
freq          196
Name: occupation, dtype: object

### Step 17. user들의 평균 나이는 몇 살인가요?
34

In [0]:
users['age'].mean()

34.05196182396607

### Step 18. 데이터셋에서 가장 적게 등장하는 나이는 몇 살인가요?

가장 적게 등장하는 나이는 1번씩 등장하는 나이이며  7, 66, 73, 10, 11살 입니다. 이러한 결과를 조회할수 있도록 데이터 조회하는 코드를 작성해주세요.

In [0]:
users.groupby('age').count().sort_values('gender')

gender  occupation  zip_code
age                              
7         1           1         1
66        1           1         1
73        1           1         1
10        1           1         1
11        1           1         1
62        2           2         2
69        2           2         2
68        2           2         2
64        2           2         2
58        3           3         3
59        3           3         3
61        3           3         3
70        3           3         3
65        3           3         3
14        3           3         3
63        3           3         3
54        4           4         4
16        5           5         5
13        5           5         5
15        6           6         6
56        6           6         6
52        6           6         6
60        9           9         9
57        9           9         9
41       10          10        10
55       11          11        11
53       12          12        12
46       12          12        12
43       13          13        13
47       14          14        14
..      ...         ...       ...
45       15          15        15
38       17          17        17
34       17          17        17
18       18          18        18
37       19          19        19
49       19          19        19
48       20          20        20
50       20          20        20
51       20          20        20
42       21          21        21
36       21          21        21
40       21          21        21
39       22          22        22
44       23          23        23
19       23          23        23
31       25          25        25
33       26          26        26
21       27          27        27
35       27          27        27
23       28          28        28
32       28          28        28
20       32          32        32
29       32          32        32
24       33          33        33
26       34          34        34
27       35          35        35
28       36          36        36
22       37          37        37
25       38          38        38
30       39          39        39

[61 rows x 3 columns]